### DISCUSSION TITLE
Expose config options as environment variables

### DISCUSSION THREAD

**Discussion by djsv23** [ThioJoe/YT-Spammer-Purge#555] [Ideas]
Following the ideas from the LTT video, I'm setting up a fleet of configs to autonomously run and report comments on my favorite channels as well as some that are particularly overrun by spammers.

Right now the way to do this is to set up an individual config file for each channel I'm monitoring and spin up a docker container that mounts that particular config, which means I have a half-dozen individual config files to maintain as new releases come out and possibly introduce changes to the config file.

Maintaining this sort of setup would be enhanced greatly if a few choice config options like the scan channel ID,  and number of recent videos could be overridden by environment variables so that a single config file can be re-used.

**Comment by ThioJoe** [comment]
First I'd have to even figure out how to do that

**Reply by djsv23** [reply]
Here's an overview on the topic: https://www.twilio.com/blog/environment-variables-python 

Basically anywhere you fetch a value from config, you'd first try to get the data from an environment variable and then if that returns none, check the config for it
